source/dagger/gameplay/platformer/character_controller_fsm.cpp

```diff
@@ -44,8 +44,7 @@ void CharacterControllerFSM::Running::Exit(CharacterControllerFSM::StateComponen
 
 void CharacterControllerFSM::Running::Run(CharacterControllerFSM::StateComponent& state_) 
 {
-	auto& [sprite, input, character] = Engine::Registry()
-		.get<Sprite, InputReceiver, platformer::PlatformerCharacter>(state_.entity);
+	auto&& [sprite, input, character] = Engine::Registry().get<Sprite, InputReceiver, platformer::PlatformerCharacter>(state_.entity);
```

Comment:
running is working, I was not sure that changing sprite.position will work.

Error:
https://stackoverflow.com/questions/17771406/c-initial-value-of-reference-to-non-const-must-be-an-lvalue